core/src/avm1/object/script_object.rs

```diff
@@ -52,6 +53,7 @@ pub struct ScriptObject<'gc>(GcCell<'gc, ScriptObjectData<'gc>>);
 #[derive(Collect)]
 #[collect(no_drop)]
 pub struct ScriptObjectData<'gc> {
+    native: Option<GcCell<'gc, NativeObject<'gc>>>,
```

Comment:
* Instead of an `Option` here, there could be a `NativeObject::None` variant, and this also quickly removes the need for `Dummy`. We would have to read thru the `GcCell` to check if it's empty, but I think that shouldn't really matter -- any fn expecting a native object type and would be reading the `GcCell` anyway. 

* I'm wondering if the `GcCell` is at the right layer, instead of doing, say, `NativeObject::TextFormat(GcCell<'gc, TextFormat>)` For most things I guess they want to be `GcCell`ed anyway, but I'm thinking of cases like `StageObject` which would end up being double-Gc'd.

Reply:
Done - This actually simplified things!